src/main/java/dobby/command/DeadlineCommand.java

```diff
@@ -0,0 +1,64 @@
+package dobby.command;
+
+import java.time.format.DateTimeParseException;
+
+import dobby.DobbyException;
+import dobby.TaskList;
+import dobby.task.Deadline;
+
+public class DeadlineCommand implements Command {
+    protected static final String USAGE = "deadline _description_ /by dd/mm/yyyy hhhh(optional)";
+
+    @Override
+    public String parseInput(TaskList tasks, String text) throws DobbyException {
+        String message = "";
+        assert text.startsWith("deadline") : "Deadline command must start with deadline";
+        try {
+            int commandLen = "deadline".length();
+            text = text.substring(commandLen + 1).trim();
+
+            if (text.indexOf("/by") < 0 && text.length() >= 1) { // non-empty description and /by missing
+                throw new DobbyException("Incorrect usage of command.\nDeadline details cannot be empty. "
+                        + "Please try again.\n  " + USAGE);
+            } else if (text.indexOf("/by") == 0) { // no description, by given
+                throw new DobbyException("Incorrect usage of command.\nDescription cannot be empty. "
+                        + "Please try again.\n  " + USAGE);
+            }
+
+            String by = text.substring(text.indexOf("/by") + 4).trim();
+            String description = text.substring(0, text.indexOf("/by") - 1).trim();
+            Deadline deadline = new Deadline(description, by);
+
+            if (by.lastIndexOf(' ') > 0) {
```

Comment:
ALthough this part is not exactly arrow headed, I think you could do sth like `by.lastIndexOf(' ') > 0 && by.substring(1 + by.lastIndexOf(' ')).length() > 4` . You might also want to abstract the boolean argument into a variable with meaningul name, so that people directly understand what you are checking here.. Anyway, amazing code!